modules/libvirt/mirror/main.tf

```
@@ -17,15 +17,13 @@ module "mirror" {
   swap_file_size      = var.swap_file_size
   ssh_key_path        = var.ssh_key_path
   roles               = ["mirror"]
-  grains              = <<EOF
-
-cc_username: ${var.base_configuration["cc_username"]}
-cc_password: ${var.base_configuration["cc_password"]}
-data_disk_device: vdb
-ubuntu_distros: [${join(", ", formatlist("'%s'", var.ubuntu_distros))}]
-use_mirror_images: ${var.base_configuration["use_mirror_images"]}
-
-EOF
+  grains = {
+    cc_username = var.base_configuration["cc_username"]
+    cc_password = var.base_configuration["cc_password"]
+    data_disk_device : vdb
```

```suggestion
    data_disk_device = "vdb"
```

I think this should not compile. Please test creating a mirror locally just to be sure.

---

Fixed